File: csharp/Antlr4cs/CSharpParserBase.cs
using Antlr4.Runtime;
using System.Collections.Generic;
using System;
using System.IO;
using System.Linq;
using System.Text.RegularExpressions;

public abstract class CSharpParserBase : Parser
{
    protected CSharpParserBase(ITokenStream input)
            : base(input)
    {
    }

    protected bool IsLocalVariableDeclaration()
    {
        var local_var_decl = this.Context as Test.CSharpParser.Local_variable_declarationContext;
        if (local_var_decl == null) return true;
        var local_variable_type = local_var_decl.local_variable_type();
        if (local_variable_type == null) return true;
        if (local_variable_type.GetText() == "var") return false;
        return true;
    }
}

File: csharp/v7/Cpp/CSharpParserBase.cpp
#include "antlr4-runtime.h"
#include "CSharpParserBase.h"
#include "CSharpParser.h"
#include <algorithm>
#include <cctype>

CSharpParserBase::CSharpParserBase(antlr4::TokenStream *input)
    : antlr4::Parser(input)
{
}

const std::vector<std::string> &CSharpParserBase::allSemanticFunctions()
{
    static const std::vector<std::string> funcs = {"IsLocalVariableDeclaration"};
    return funcs;
}

void CSharpParserBase::initSemantics()
{
    if (initialized_)
        return;
    initialized_ = true;
    // NOTE: argc/argv are not available from the parser base class.
    // The test harness (Test.cpp) should call noSemantics_.insert() after
    // construction if --no-semantics was passed on the command line.
}

bool CSharpParserBase::IsLocalVariableDeclaration()
{
    initSemantics();
    if (noSemantics_.count("IsLocalVariableDeclaration"))
        return true;

    auto *ctx = dynamic_cast<CSharpParser::Local_variable_declarationContext *>(getRuleContext());
    if (ctx == nullptr)
        return true;
    auto *local_variable_type = ctx->local_variable_type();
    if (local_variable_type == nullptr)
        return true;
    if (local_variable_type->getText() == "var")
        return false;
    return true;
}
